src/dds_manager.cpp

```diff
@@ -953,9 +962,29 @@ bool DDSManager::replaceFilter(const std::string& topicName,
         dynamic_cast<DDS::ContentFilteredTopic*>
         (dataReader->get_topicdescription());
```

Comment:
This appears to leak a reference count (existing code, before this PR).

IDL-defined operations that return strings or object references (or some other types) depend on the caller to take responsibility for the ownership (string) or reference count (object).  

Most often, this is handled by directly assigning the result to a local variable of the corresponding `*_var` type.  In some special cases, the reference is returned directly to the caller of this function (or equivalent).

Reply:
Looks like we do something similar to this in 2 places.  What is the suggested fix?

Reply:
It's based on the OMG specs that were current at the time OpenDDS was developed, it was not a decision made by the OpenDDS developers (except the decision to use the specs).

OpenDDS v4 will use a newer language mapping that fixes this issue.

Reply:
```
DDS::TopicDescription_var topic = dataReader->get_topicdescription();
DDS::ContentFilteredTopic_var topicDesc = DDS::ContentFilteredTopic::_narrow(topic);
```

Each IDL local interface type has a static `_narrow()` function that works a lot like the C++11 standard library's `dynamic_pointer_cast` -- it does a dynamic_cast but also knows about the reference counting smart pointer.

Reply:
OK.  Thanks.  Should be all good to go now.